src/go/rpk/pkg/cli/container/purge.go

```diff
@@ -12,43 +12,69 @@ package container
 import (
 	"context"
 	"fmt"
+	"os"
 	"sync"
 
 	"github.com/docker/docker/api/types"
 	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/container/common"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/profile"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
 	"golang.org/x/sync/errgroup"
 )
 
-func newPurgeCommand() *cobra.Command {
+func newPurgeCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 	command := &cobra.Command{
 		Use:   "purge",
 		Short: "Stop and remove an existing local container cluster's data",
-		RunE: func(_ *cobra.Command, _ []string) error {
-			c, err := common.NewDockerClient()
+		Run: func(cmd *cobra.Command, _ []string) {
+			c, err := common.NewDockerClient(cmd.Context())
+			out.MaybeDie(err, "unable to create docker client: %v", err)
+			defer c.Close()
+
+			purged, err := purgeCluster(c)
+			out.MaybeDieErr(common.WrapIfConnErr(err))
+
+			if !purged {
+				return
+			}
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			y, ok := cfg.ActualRpkYaml()
+			if !ok || y.Profile(common.ContainerProfileName) == nil {
+				// rpk.yaml file nor profile exist, we exit.
+				return
+			}
+			cleared, err := profile.DeleteProfile(fs, y, common.ContainerProfileName)
 			if err != nil {
-				return err
+				fmt.Fprintf(os.Stderr, "unable to delete %q profile: %v; you may delete the profile manually running 'rpk profile delete %v'", common.ContainerProfileName, err, common.ContainerProfileName)
+				return
+			}
+			fmt.Printf("Deleted %q profile.\n", common.ContainerProfileName)
```

Comment:
move `profile` in front of `%q` here -- this will also match the `rpk profile delete` command output.